### ISSUE TITLE
[import/consistent-type-specifier-style] Autofix creates syntax errors and unprettified results

### ISSUE THREAD

**Issue by KirillGalimov** [import-js/eslint-plugin-import#2790]
<ins>**Config:**</ins>
```typescript
'import/consistent-type-specifier-style': ['error', 'prefer-top-level']
```

<ins>**Case 1:**</ins>

input
```typescript
import React, { createContext, type Reducer } from 'react';

export const Component = ({ propOne, propTwo }) => {
  return null;
};
```

output
```typescript
import React, { createContext  } from 'react'; // unnecessary space added
import type {Reducer} from 'react'; // necessary spaces not added

export const Component = ({ propOne propTwo }) => { // comma removed (syntax error)
  return null;
};
```

<ins>**Case 2:**</ins>

input
```typescript
import React, { createContext, type Reducer } from 'react';
import { useLocation, useHistory } from 'react-router';

export const Component = ({ propOne, propTwo }) => {
  return null;
};
```

output
```typescript
import React, { createContext  } from 'react'; // unnecessary space added
import type {Reducer} from 'react'; // necessary spaces not added
import { useLocation useHistory } from 'react-router'; // comma removed (syntax error)

export const Component = ({ propOne, propTwo }) => {
  return null;
};
```

**Comment by makuko** [comment]
I noticed the same behavior. It removes commas in random places unrelated to imports.

**Comment by ljharb** [comment]
This is a major bug, thanks for reporting it.

"unprettified results" are never a bug; formatting is expected to be handled by other rules - but producing a syntax error is unacceptable.

**Comment by aaronadamsCA** [comment]
Note this seems to be a duplicate of #2753, fixed in #2754; the fix remains unreleased.

**Comment by ljharb** [comment]
Thanks! Closed as a duplicate of #2753.

I'll try to get a release out soon.
